src/Arcus.Testing.Tests.Unit/Core/PollTests.cs

```diff
@@ -0,0 +1,236 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Bogus;
+using Bogus.Extensions;
+using Polly.Timeout;
+using Xunit;
+
+namespace Arcus.Testing.Tests.Unit.Core
+{
+    public class PollTests : IAsyncLifetime
+    {
+        private readonly TimeSpan _100ms = TimeSpan.FromMilliseconds(100), _10ms = TimeSpan.FromMilliseconds(10);
+        private readonly object _expectedResult = Bogus.PickRandom((object) Bogus.Random.Int(), Bogus.Random.AlphaNumeric(10));
+
+        private static int Index;
+        private static readonly Faker Bogus = new();
+
+        [Fact]
+        public async Task Poll_WithTargetAvailableWithinTimeFrame_SucceedsByContinuing()
```

Comment:
Wanted to create dedicated types and models for this, but figured that since this is Arcus and we should also consider juniors, it might be best to test it like this: overloads categorized by descriptive test methods.